lib/MongoDB/Role/_ClusterTimeModifier.pm

```diff
@@ -0,0 +1,87 @@
+#
+#  Copyright 2014 MongoDB, Inc.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+
+use strict;
+use warnings;
+package MongoDB::Role::_ClusterTimeModifier;
+
+# MongoDB role to manage clusterTime commands
+
+use Moo::Role;
+
+use MongoDB::Error;
+use MongoDB::_Types -types, 'to_IxHash';
+use Scalar::Util qw/ blessed /;
+
+use namespace::clean;
+
+# As cluster time and sessions are hand in hand, makes sense to apply them here
+with $_ for qw(
+  MongoDB::Role::_SessionModifier
+);
+
+requires qw/client session/;
```

Comment:
I think we can avoid direct need of client as a method -- we can always go through the session to get to the client.  If there is no session (explicit or implicit) then the topology doesn't support clusterTime, either and we can shortcut.